### ISSUE TITLE
cannot offset menulist (popperjs)

### ISSUE THREAD

**Issue by hesxenon** [chakra-ui/chakra-ui#2055]
# Bug report

## Describe the bug

Offsetting the menu list as described in popperjs docs (and subsequent types) results in console error

## To reproduce

1. Add an offset modifier to the `<Menu>` tag
2. Console logs error like `cannot read property 'y' of undefined

## Minimal reproduction

https://codesandbox.io/s/happy-volhard-pup56?file=/src/App.js

## Expected behavior

I'd expect to be able to offset the menu list without an error

## Additional context

Seems like the props are spread correctly through the layers, i.e. passed correctly to `usePopper`, but the `offsetData` from the `modifierState` is undefined at first.

**Comment by segunadebayo** [comment]
Hi @hesxenon,

I can't seem to see a useful reproduction in the attached link. Could you update that?

Could you try passing the `gutter` prop to the <Menu> to see if it solves your use-case? Let us know,

**Comment by hesxenon** [comment]
sorry, I've updated the sandbox and found the error along the way. When you look at [popperjs offset modifier docs](https://popper.js.org/docs/v2/modifiers/offset/) and set `modifiers` like they state, it works as expected.

_However_ the typings require you to specify the `fn` prop on a modifier object, and this breaks the functionality. If I remember correctly, you're just re-exporting the types from popperjs, right? If so, this seems like a popperjs issue, and not a chakra issue, though a type fix from your side would be appreciated.

**Comment by segunadebayo** [comment]
Yes, it's mostly an error from popper.js. However, I've just pushed a fix to support the `gutter` prop for the tooltip. You either pass the gutter prop, (as in `gutter={8}`) or pass offset prop (as in `offset=[20, 50]`).

Regarding the type issue, I think it's also an issue from popper.js 'coz we faced some challenges here as well. Feel free to open an issue with popper.js if you still face this type of issue.

Cheers.
